### PR TITLE
web/resource: introduce RouteParams and a Resolve() variant

### PR DESCRIPTION
**Pull request by amery**

### CODE REVIEW

**Comment by amery** [comment]
depends on #45, waiting to rebase.

**Review by karasz — COMMENTED** [review]
I see nothing wrong with this PR

File: web/resource/resolve.go [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+// Len tells how many parameters are stored in the table
+func (params RouteParamsTable) Len(key string) (int, bool) {
+	if m, ok := params[key]; ok {
+		return len(m), true
+	}
+	return 0, false
+}
+
+// Get returns the first stored value for a parameter
+func (params RouteParamsTable) Get(key string) (any, bool) {
```
Comment (karasz) [inline web/resource/resolve.go:64]: `First`, `Last` and `All` maybe better than `Get*`?

**Review by karasz — COMMENTED** [review]
Other than the previous comment I am fine with this

File: web/resource/resolve.go (hunk 1 above)
Comment (amery) [reply web/resource/resolve.go]: amended accordingly

**Comment by amery** [comment]
amended as recommended

**Review by karasz — APPROVED** [review]
+1

**Comment by amery** [comment]
> +1

`darvaza.org/x/web@v0.6.2` tagged

**Pull request merged**